.travis.yml

```diff
@@ -22,26 +22,32 @@ addons:
     packages:
       - libgconf-2-4
 
-script:
-  - npm run test:coverage
-  - npm run build
-  - npm run sonar
-
-after_script:
-  - npm run test:coveralls
-  - npm run test:acceptance
-
-deploy:
-  - provider: npm
-    skip_cleanup: true
-    email: $NPM_MAIL
-    api_key: $NPM_TOKEN
-    on:
-      tags: true
-
-  - provider: script
-    # Workaround. cannot login in NPM and cannot configure NPM provider to go in the directory
-    script: cd test-cypress/ && echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc && npm publish
-    skip_cleanup: true
-    on:
-      tags: true
+jobs:
+  include:
+    - stage: build and test
+      if: type = pull_request
```

Comment:
This would skip any builds for master, effectively skipping sonar analyses for master which would lead to incorrect analyses for merge requests. Maybe add another stage with test (with coverage) and sonar analysis.

Also, I'm taking for master but it's applicable for release branches as well.  However I don't see them analysed so maybe it's won't be a problem for them.